lib/modules/zfs.go

```diff
@@ -0,0 +1,60 @@
+package modules
+
+import (
+	"bufio"
+	"fmt"
+	"github.com/davidscholberg/go-i3barjson"
+	"os/exec"
+	"strings"
+)
+
+// Zfs represents the configuration data for the ZFS block
+type Zfs struct {
+	BlockConfigBase `yaml:",inline"`
+	PoolName        string `yaml:"zpool_name"`
+	ZpoolBin        string `yaml:"zpool_bin"`
+}
+
+// UpdateBlock updates the ZFS block
+func (c Zfs) UpdateBlock(b *i3barjson.Block) {
+	b.Color = c.Color
+	fullTextFmt := fmt.Sprintf("%s%s - %%s", c.Label, c.PoolName)
+
+	zpoolCmd := exec.Command("sudo", c.ZpoolBin, "status", c.PoolName)
+	out, err := zpoolCmd.StdoutPipe()
```

Comment:
Using `cmd.StdoutPipe()` is only really necessary if you expect a large volume of output. In this case, the output is only going to be a few short lines, so it would simplify the code if you just used `cmd.Output()`.